src/main/mbedtls.c

```diff
@@ -0,0 +1,27 @@
+/**
+ * @file mbedtls.c  MbedTLS initialisation and multi-threading routines
+ *
+ */
+#ifdef HAVE_SIGNAL
+#include <signal.h>
+#endif
+#include "main.h"
+
+
+#ifdef SIGPIPE
+static void sigpipe_handler(int x)
+{
+	(void)x;
+	(void)signal(SIGPIPE, sigpipe_handler);
+}
+#endif
+
+
+int mbedtls_init(void)
+{
+#ifdef SIGPIPE
+	(void)signal(SIGPIPE, sigpipe_handler);
+#endif
```

Comment:
is this part also needed for MbedTLS ?

Reply:
I believe it's not a OpenSSL-specific workaround, and it's needed for MbedTLS as well https://stackoverflow.com/questions/25144550/is-it-possible-to-force-openssl-to-not-generate-sigpipe-without-global-signal-ha